src/game/common/statemachine.cpp

```diff
@@ -0,0 +1,716 @@
+    }
+
+    captainslog_dbgassert(status <= STATE_CONTINUE, "Please handle sleep states prior to this");
+
+    switch (status) {
+        case STATE_FAILURE: {
+            if (m_failureStateID == EXIT_MACHINE_WITH_SUCCESS) {
+                Get_Machine()->Internal_Set_State(MACHINE_DONE_STATE_ID);
+                return STATE_SUCCESS;
+            } else if (m_failureStateID == EXIT_MACHINE_WITH_FAILURE) {
+                Get_Machine()->Internal_Set_State(MACHINE_DONE_STATE_ID);
+                return STATE_FAILURE;
+            } else {
+                return Get_Machine()->Internal_Set_State(m_failureStateID);
+            }
+        } break;
+
+        case STATE_SUCCESS: {
+            if (m_successStateID == EXIT_MACHINE_WITH_SUCCESS) {
+                Get_Machine()->Internal_Set_State(MACHINE_DONE_STATE_ID);
+                return STATE_SUCCESS;
+            } else if (m_successStateID == EXIT_MACHINE_WITH_FAILURE) {
+                Get_Machine()->Internal_Set_State(MACHINE_DONE_STATE_ID);
+                return STATE_FAILURE;
+            } else {
+                return Get_Machine()->Internal_Set_State(m_successStateID);
+            }
+        } break;
+
+        case STATE_CONTINUE: {
+            if (!!m_transitions.empty()) {
+                for (auto &info : m_transitions) {
+                    if (info.test(this, info.user_data)) {
+#ifdef GAME_DEBUG_STRUCTS
+                        if (Get_Machine()->Get_Wants_Debug_Output()) {
+                            const char *description;
+
+                            if (info.description) {
+                                description = info.description;
+                            } else {
+                                description = "[no description]";
+                            }
+
+                            captainslog_debug("%d '%s' -- '%s' condition '%s' returned true!",
+                                g_theGameLogic->Get_Frame(),
+                                Get_Machine_Owner()->Get_Template()->Get_Name().Str(),
+                                Get_Machine()->Get_Name().Str(),
+                                description);
+                        }
+#endif
+                        if (info.to_state_id == EXIT_MACHINE_WITH_SUCCESS) {
+                            return STATE_SUCCESS;
+                        } else if (info.to_state_id == EXIT_MACHINE_WITH_FAILURE) {
+                            return STATE_FAILURE;
+                        } else {
+                            return Get_Machine()->Internal_Set_State(info.to_state_id);
+                        }
+                    }
+                }
+            }
+        } break;
+    }
+
+    return STATE_CONTINUE;
+}
+
+StateMachine::StateMachine(Object *owner, Utf8String name) :
+    m_owner(owner),
+    m_updateFrame(0),
+    m_defaultStateID(INVALID_STATE_ID),
+    m_defaultStateInited(false),
+    m_currentState(nullptr),
+    m_locked(0)
+#ifdef GAME_DEBUG_STRUCTS
+    ,

```

Comment:
Uh. Is this how clang-format formats this? Looks really ugly.

Reply:
Yeah that's how its formatted.